### ISSUE TITLE
3dView not working with grayscale 8-bit input images

### ISSUE THREAD

**Issue by jcbenoit** [PetteriAimonen/focus-stack#24]
Task_3DPreview::task() supports only input color RGB images.
Would be great to add support for input grayscale 8-bit images.
It just need to add a switch instruction, and handle the different images types.

for grayscale 8-bit: case:
replace line
cv::Vec3b pixel = merged.at<cv::Vec3b>(y, x);
by  
uchar pixel = merged.at<uchar>(y, x);

and replace  lines
m_result.at<cv::Vec4b>(cam_y_d, cam_x) = cv::Vec4b(pixel[0], pixel[1], pixel[2], 255);
m_result.at<cv::Vec4b>(cam_y_d, cam_x + 1) = cv::Vec4b(pixel[0], pixel[1], pixel[2], 255);
by
m_result.at<cv::Vec4b>(cam_y_d, cam_x) = cv::Vec4b(pixel, pixel, pixel, 255);
m_result.at<cv::Vec4b>(cam_y_d, cam_x + 1) = cv::Vec4b(pixel, pixel, pixel, 255);

**Comment by PetteriAimonen** [comment]
Makes sense. If you can make a pull request I'll merge it, otherwise I'll add this myself once I have time.
